src/components/BlogForm/BlogForm.jsx

```diff
@@ -1,23 +1,45 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
 import { Form, Field } from "react-final-form";
 import { useDispatch } from "react-redux";
-import { addBlogPostInLocalStorage } from "@actions/blogActions";
+import {
+  addBlogPostInLocalStorage,
+  editBlogPostInLocalStorage,
+} from "@actions/blogActions";
 import ImageDnD from "@components/ImageDnD/ImageDnD";
 import SelectBox from "@components/SelectBox/SelectBox";
 import tags from "@constants/tags.json";
 import "@components/BlogForm/blogForm.scss";
 import Button from "@components/Button/Button";
 
-const BlogForm = () => {
+const BlogForm = ({ initialData, onSubmit }) => {
   const dispatch = useDispatch();
   const [imagePreview, setImagePreview] = useState(null);
   const [selectedTags, setSelectedTags] = useState([]);
+  const [isEditMode, setIsEditMode] = useState(false);
 
-  const onSubmit = (values, form) => {
-    const blog = { ...values, image: imagePreview, tags: selectedTags };
-    dispatch(addBlogPostInLocalStorage(blog));
-    setImagePreview(null);
+  useEffect(() => {
+    if (initialData) {
+      setImagePreview(initialData.image);
+      setSelectedTags(initialData.tags);
+      setIsEditMode(true);
+    }
+  }, [initialData]);
+
+  const handleSubmit = (values, form) => {
+    const blog = {
+      ...values,
+      image: imagePreview,
+      tags: selectedTags,
+    };
+    if (isEditMode) {
+      dispatch(editBlogPostInLocalStorage(initialData.id, blog));
+    } else {
+      dispatch(addBlogPostInLocalStorage(blog));
```

Comment:
```suggestion
      dispatch(addBlog(blog));
```
Also change it where it is used.

Reply:
done